### PR TITLE
Fixed TOCs

### LINKED ISSUE #514: Articles not in the TOC
There seems to be quite a lot of articles that are not part of [the site TOC](https://github.com/dotnet/core-docs/blob/master/docs/toc.md). Specifically (excluding 🔧 article stubs and TOC articles):

* [async/async-csharp-vb.md](https://dotnet.github.io/docs/async/async-csharp-vb.html): This seems to be a copy of [languages/csharp/async.md](https://dotnet.github.io/docs/languages/csharp/async.html), and a remnant of #175. Probably can be deleted.
* [async/async-fsharp.md](https://dotnet.github.io/docs/async/async-fsharp.html): Maybe put it under Learning F#? Or delete it, unless more coverage on F# is planned.
* [async/async-overview.md](https://dotnet.github.io/docs/async/async-overview.html): Short article, almost a TOC, maybe include into C#/F# async articles?
* [concepts/assembly-format.md](https://dotnet.github.io/docs/concepts/assembly-format.html), [concepts/class-libraries.md](https://dotnet.github.io/docs/concepts/class-libraries.html), [concepts/common-type-system.md](https://dotnet.github.io/docs/concepts/common-type-system.html), [concepts/delegates-lambdas.md](https://dotnet.github.io/docs/concepts/delegates-lambdas.html), [concepts/framework-libraries.md](https://dotnet.github.io/docs/concepts/framework-libraries.html), [concepts/gc-overview.md](https://dotnet.github.io/docs/concepts/gc-overview.html), [concepts/generics.md](https://dotnet.github.io/docs/concepts/generics.html), [concepts/linq.md](https://dotnet.github.io/docs/concepts/linq.html), [concepts/managed-code.md](https://dotnet.github.io/docs/concepts/managed-code.html), [concepts/native-interop.md](https://dotnet.github.io/docs/concepts/native-interop.html): Since [concepts/index.md](https://dotnet.github.io/docs/concepts/index.html) is the top level Learn .NET section, these articles probably belong directly under it.
* [core-concepts/core-sdk/sdk-overview.md](https://dotnet.github.io/docs/core-concepts/core-sdk/sdk-overview.html), [core-concepts/core-sdk/cli/dotnet-test-protocol.md](https://dotnet.github.io/docs/core-concepts/core-sdk/cli/dotnet-test-protocol.html): Their parents have a TOC entry, so these belong under them.
* [essentials/numericsincore.md](https://dotnet.github.io/docs/essentials/numericsincore.html): Already handled by #507.
* [essentials/exceptions/exception-class-properties.md](https://dotnet.github.io/docs/essentials/exceptions/exception-class-properties.html), [essentials/exceptions/exception-hiearchy.md](https://dotnet.github.io/docs/essentials/exceptions/exception-hiearchy.html), [essentials/exceptions/exceptions-best-practices.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-best-practices.html), [essentials/exceptions/exceptions-catch-specific-exceptions.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-catch-specific-exceptions.html), [essentials/exceptions/exceptions-create-user-defined.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-create-user-defined.html), [essentials/exceptions/exceptions-throw-exceptions.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-throw-exceptions.html), [essentials/exceptions/exceptions-use-finally-blocks.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-use-finally-blocks.html), [essentials/exceptions/exceptions-use-try-catch-block.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-use-try-catch-block.html), [essentials/exceptions/exceptions-user-filtered-handlers.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-user-filtered-handlers.html), [essentials/exceptions/exceptions.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions.html): These belong under [essentials/exceptions.md](https://dotnet.github.io/docs/essentials/exceptions.html) which is already in the TOC.
* [essentials/exceptions/exceptions-com-interop.md](https://dotnet.github.io/docs/essentials/exceptions/exceptions-com-interop.html): Should be deleted per https://github.com/dotnet/core-docs/pull/344#discussion_r57095278.
* [languages/csharp/delegate-class.md](https://dotnet.github.io/docs/languages/csharp/delegate-class.html), [languages/csharp/delegates-overview.md](https://dotnet.github.io/docs/languages/csharp/delegates-overview.html), [languages/csharp/delegates-patterns.md](https://dotnet.github.io/docs/languages/csharp/delegates-patterns.html), [languages/csharp/delegates-strongly-typed.md](https://dotnet.github.io/docs/languages/csharp/delegates-strongly-typed.html), [languages/csharp/distinguish-delegates-events.md](https://dotnet.github.io/docs/languages/csharp/distinguish-delegates-events.html), [languages/csharp/event-pattern.md](https://dotnet.github.io/docs/languages/csharp/event-pattern.html), [languages/csharp/events-overview.md](https://dotnet.github.io/docs/languages/csharp/events-overview.html), [languages/csharp/modern-events.md](https://dotnet.github.io/docs/languages/csharp/modern-events.html): Put under [languages/csharp/delegates-events.md](https://dotnet.github.io/docs/languages/csharp/delegates-events.html) and remove Previous and Next links from articles.
* [languages/csharp/expression-classes.md](https://dotnet.github.io/docs/languages/csharp/expression-classes.html), [languages/csharp/expression-trees-building.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-building.html), [languages/csharp/expression-trees-execution.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-execution.html), [languages/csharp/expression-trees-explained.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-explained.html), [languages/csharp/expression-trees-interpreting.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-interpreting.html), [languages/csharp/expression-trees-summary.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-summary.html), [languages/csharp/expression-trees-translating.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-translating.html): Put under [languages/csharp/expression-trees.md](https://dotnet.github.io/docs/languages/csharp/expression-trees.html) and remove Previous and Next links.
* [languages/csharp/implicitly-typed-lambda-expressions.md](https://dotnet.github.io/docs/languages/csharp/implicitly-typed-lambda-expressions.html): This is a footnote for [languages/csharp/expression-trees-interpreting.md](https://dotnet.github.io/docs/languages/csharp/expression-trees-interpreting.html#examining-a-simple-addition-expression), and should stay outside the TOC.
* [project-model/project-json-reference.md](https://dotnet.github.io/docs/project-model/project-json-reference.html): Probably put directly under .Net Core concepts for now, since the other project-model articles are 🔧stubs and not in the TOC.
* [tutorials/getting-started-with-csharp/console-teleprompter.md](https://dotnet.github.io/docs/tutorials/getting-started-with-csharp/console-teleprompter.html), [tutorials/getting-started-with-csharp/console-webapiclient.md](https://dotnet.github.io/docs/tutorials/getting-started-with-csharp/console-webapiclient.html), [tutorials/getting-started-with-csharp/microservices.md](https://dotnet.github.io/docs/tutorials/getting-started-with-csharp/microservices.html), [tutorials/getting-started-with-csharp/working-with-linq.md](https://dotnet.github.io/docs/tutorials/getting-started-with-csharp/working-with-linq.html): Put under [tutorials/index.md](https://dotnet.github.io/docs/tutorials/index.html).

Do you agree with my assessment of what to do with these?

### PR DESCRIPTION
**Pull request by svick**
Fixes #514

### CODE REVIEW

File: docs/essentials/collections/collections-and-data-structures.md [hunk 1]
```diff
@@ -55,13 +55,13 @@ Manage efficient storage and access of unique elements | [System.Collections.Gen
...
+[When to Use Generic Collections](when-to-use-generic-collections.md) | Discusses the use of generic collection types.
+[Comparisons and Sorts Within Collections](comparisons-and-sorts-within-collections.md) | Discusses the use of equality comparisons and sorting comparisons in collections.
+[Sorted Collection Types](sorted-collection-types.md) | Describes sorted collections performance and characteristics.
+[Hashtable and Dictionary Collection Types](hashtable-and-dictionary-collection-types.md) | Describes the features of generic and non-generic hash-based dictionary types.
+[Thread-Safe Collections](thread-safe-collections.md) | Describes collection types such as [System.Collections.Concurrent.BlockingCollection&lt;T&gt;](http://dotnet.github.io/api/System.Collections.Concurrent.BlockingCollection%601.html) and [System.Collections.Concurrent.ConcurrentBag&lt;T&gt;](http://dotnet.github.io/api/System.Collections.Concurrent.ConcurrentBag%601.html) that support safe and efficient concurrent access from multiple threads.
```
Comment (stevehoag) [inline docs/essentials/collections/collections-and-data-structures.md]: Doesn't renaming these files break all existing links?

Comment (mairaw) [inline docs/essentials/collections/collections-and-data-structures.md]: I think he fixed those too @stevehoag. when you run docfx, it notifies you of broken links.

Comment (svick) [inline docs/essentials/collections/collections-and-data-structures.md]: @stevehoag Yeah, I fixed all the links inside the docs. docfx, does not report any link errors.

**Comment by mairaw** [comment]
Thanks for you thorough review @svick. This fixes a lot of the current issues we have with missing links in the TOC. I'd like some other folks to review these since we've been talking about the TOC this week.

/cc @richlander @blackdwarf

**Comment by richlander** [comment]
LGTM

Let's get this in before any other TOC changes occur.

**Pull request merged**